packages/go/project.bri

```diff
@@ -80,13 +99,20 @@ export async function goInstall(
   const modules = goModDownload(options.goModule);
 
   let buildResult = std.runBash`
-    go install
+    # Run generate if requested
+    if [ "${options.buildParams?.generate}" = "true" ]; then
+      go generate ./...
+    fi
+
+    go install -ldflags="$ldflags" $package_path
   `
     .workDir(options.goModule)
     .dependencies(go())
     .env({
       GOMODCACHE: modules,
       GOBIN: std.tpl`${std.outputPath}/bin`,
+      ldflags: options.buildParams?.ldflags?.join(" ") ?? "",
```

Comment:
very small and subtle issue, but this won't work correctly if any arg in `ldflags` has spaces in it. According to the Go docs (https://go.dev/cmd/go/?m=old):

> The -asmflags, -gccgoflags, -gcflags, and -ldflags flags accept a space-separated list of arguments to pass to an underlying tool during the build. To embed spaces in an element in the list, surround it with either single or double quotes. The argument list may be preceded by a package pattern and an equal sign, which restricts the use of that argument list to the building of packages matching that pattern [...]

I think there are two directions that would make sense to me:

1. Change `ldflags` to a single string, and let the user worry about argument escaping and the package name
2. Keep `ldflags` as an array, and ensure it gets passed properly. This would mean wrapping args in single quotes or double quotes as needed (and there are of course corner cases when an argument has single quotes, double quotes, and an equals sign)
   - A short-term option would be to just add an assert that fails if any argument contains `'`/`"`/`=` to leave it as a future to-do

Reply:
I updated it in the last commit by adding a wrapper to the array containing the ldflags. Basically, I'm escaping single and double quotes as you suggested, but not the =. I had some issues when I tried to escape this special character:

```bash
[58353] /home/brioche-runner-fa2012085166c82d3f8df96aec04cca374bf4df217df39a37b78d082950263da/.local/share/brioche/locals/761f9df90da51dbae24c4c20e4c8ce338a35604cb39c
e4fa611c6c556008d46f/go/pkg/tool/linux_amd64/link: -X flag requires argument of the form importpath.name=value
```

Find below the fixtures I used for the ldflags to verify the escaping was working as expected while building #55:

- ["-s", "-w", \`-X main.version=${project.version}`]
- ["-s", "-w", \`-X 'main.version=${project.version}'`]
- ["-s", "-w", \`-X "main.version=${project.version}"`]
- ["-s", "-w", "-X", \`main.version=${project.version}`]
- ["-s", "-w", "-X", \`'main.version=${project.version}'`]
- ["-s", "-w", "-X", \`"main.version=${project.version}"`]